tools/codegen/codespec/api_spec_schema.go

```diff
@@ -0,0 +1,63 @@
+package codespec
+
+import (
+	"fmt"
+
+	"github.com/pb33f/libopenapi/datamodel/high/base"
+	high "github.com/pb33f/libopenapi/datamodel/high/v3"
+)
+
+var errInvalidSchema = fmt.Errorf("invalid schema")
+
+type APISpecSchema struct {
+	Schema *base.Schema
+	Type   string
+}
+
+type APISpecResource struct {
+	Description        *string
+	DeprecationMessage *string
+	CreateOp           *high.Operation
+	ReadOp             *high.Operation
+	UpdateOp           *high.Operation
+	DeleteOp           *high.Operation
+	CommonParameters   []*high.Parameter
+}
+
+func (s *APISpecSchema) GetComputability(name string) ComputedOptionalRequired {
+	for _, prop := range s.Schema.Required {
```

Comment:
if s.Schema.Required is a []string,   slices.Contains

Reply:
done, thanks for the tip!